src/ui/TemplateMessageItemBody/index.tsx

```diff
@@ -0,0 +1,73 @@
+import './index.scss';
+import React, { ReactElement } from 'react';
+import type { BaseMessage } from '@sendbird/chat/message';
+import { getClassName } from '../../utils';
+import MessageTemplateProvider from '../../modules/GroupChannel/components/MessageTemplateProvider';
+import { MessageTemplateData, MessageTemplateItem, SendbirdMessageTemplate } from './types';
+import restoreNumbersFromMessageTemplateObject from './utils/restoreNumbersFromMessageTemplateObject';
+import mapData from './utils/mapData';
+import selectColorVariablesByTheme from './utils/selectColorVariablesByTheme';
+import { SendbirdTheme } from '../../types';
+
+interface Props {
+  className?: string | Array<string>;
+  message: BaseMessage;
+  isByMe?: boolean;
+  mouseHover?: boolean;
+  isReactionEnabled?: boolean;
+  theme?: SendbirdTheme;
+}
+
+/**
+ * Returns copied message template object filled with given template data and color variables.
+ */
+const getFilledMessageTemplateWithData = (
+  template: MessageTemplateItem[],
+  templateData: Record<string, any>,
+  colorVariables: Record<string, string>,
+  theme: SendbirdTheme,
+): MessageTemplateItem[] => {
```

Comment:
Isn't it better to keep the parameter type to object?

Reply:
you mean
```
templateData: object,
colorVariables: object,
```
?
I think this ok also